src/js/send-web-vitals.js

```diff
@@ -80,32 +128,32 @@ function sendWebVitals() {
       prefersColorScheme = 'not supported';
     }
 
+    const params = Object.assign({
+      event_category: 'Web Vitals',
+      event_value: Math.round(name === 'CLS' ? delta * 1000 : delta),
+      event_label: id,
+      nonInteraction: true,
+
+      effective_type: effectiveType,
+      data_saver: dataSaver,
+      device_memory: deviceMemory,
+      prefers_reduced_motion: prefersReducedMotion,
+      prefers_color_scheme: prefersColorScheme,
+      navigation_type: navigationType,
+    }, overrides);
 
-    gtag('event', name, Object.assign(
-      {
-        event_category: 'Web Vitals',
-        event_value: Math.round(name === 'CLS' ? delta * 1000 : delta),
-        event_label: id,
-        nonInteraction: true,
-
-        // See: https://web.dev/debug-web-vitals-in-the-field/
-        // UA - Remove once fully migrated to GA4
-        dimension1: overrides.debug_target,
-        dimension2: effectiveType,
-        dimension3: dataSaver,
-        dimension4: deviceMemory,
-        dimension5: prefersReducedMotion,
-        dimension6: prefersColorScheme,
-        dimension7: navigationType,
-        //GA4
-        effective_type: effectiveType,
-        data_saver: dataSaver,
-        device_memory: deviceMemory,
-        prefers_reduced_motion: prefersReducedMotion,
-        prefers_color_scheme: prefersColorScheme,
-        navigation_type: navigationType,
-      }, overrides));
+    gtag('event', name, params);
+
+  }
 
+  // Monitor LoAFs
+  if (PerformanceObserver.supportedEntryTypes.includes('long-animation-frame')) {
+    new PerformanceObserver(entries => {
+      longAnimationFrames = longAnimationFrames.concat(entries.getEntries());
```

Comment:
We never clear down the `longAnimationFrames` value. That means this will just grow and grow and grow as long as the user remains on the page. At the same time, you cannot clear down until you know the INP entry (which won't be reported until later), so tricky. This is one of the complications I was talking about.

Would you be open to additionally monitoring all long LoAFs (>= 150ms)? Or all long LoAFs with interactions? That way we could compare the two approaches.

Reply:
An alternative approach is to not use a Performance Observer for the LoAFs and just look them up in the performance timeline:

```
performance.getEntriesByType('long-animation-frame');
```

Now, they aren't guaranteed to hang round forever (which is why PO is generally preferred), so may get some blank entries for long lived pages, but easier than managing this in our own JavaScript.

Reply:
For now I was planning to monitor the `debug_loaf_length` value and take action if it grows beyond a reasonable amount for most users. If that works for you we can pick the threshold.

Reply:
So for now the [Performance Timeline buffer is set to 200 in Chrome](https://github.com/chromium/chromium/blob/54a5d0c487f5d388613f0b3fca30dcb05a0a5163/third_party/blink/renderer/core/timing/performance.cc#L256) so could go with that, which should surely be enough for most pages.

Then again if doing that, then why just use `performance.getEntriesByType('long-animation-frame');`?

Reply:
Didn't see your latest comment before reply. Yes that seems like a much better solution.

Reply:
Added that to the LoAF post too.

Reply:
PTAL